app/services/geocode_api_client.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'uri'
+require 'json'
+
+# get latitude and longitude for city
+class GeocodeApiClient
+  def initialize(city:, state:)
+    @city = city
+    @state = state
+    @base_url = ENV['GEOCODE_API_BASE_URL']
+    @api_key = ENV['GEOCODE_API_KEY']
+  end
+
+  def fetch_city_coordinates
+    response = Net::HTTP.get_response(build_uri)
+    raise "API request failed with status code: #{response.code}" unless response.is_a?(Net::HTTPSuccess)
```

Comment:
I feel like these, error messages should be in some constants.rb or some errors.rb file